File: FizzBuzz/TDD/java/src/main/java/PrinterJob.java
import java.util.List;
import java.util.Arrays;

enum PrinterJob {
  $;

  static String print(final Object value) {
      if (value instanceof List<?>) {
	  return print((List<String>) value);
      }
      if(value instanceof String) {
	  return (String) value;
      }
      throw new UnexpectedObjectTypeException(value);
  }

  static String print(final List<String> list) {
      final List<Printer> printers = PrintersProvider.all();
      if (!printers.isEmpty()) {
	  final Printer firstPrinter = printers.get(0);
	  if(firstPrinter != null)
	      {
		  return firstPrinter.print(list);
	      }
      }
      return DefaultPrinter.getInstance().print(list);
  }
}

File: FizzBuzz/TDD/JavaScript/src/PrinterJob.js
const printersProvider = require('./PrintersProvider').printersProvider

printFun = (result) => {
    if(result instanceof Array) {
	const printer = printersProvider.available()
	const resultAsString = Array.from(result).join(printer()).trim(' ')
	return resultAsString
    }
    return result
}

module.exports = {printerJob: { process: printFun } }
